src/paho/mqtt/client.py

```diff
@@ -567,16 +585,60 @@ def __init__(self, client_id="", clean_session=True, userdata=None,
         self._on_publish = None
         self._on_unsubscribe = None
         self._on_disconnect = None
+        self._on_socket_open = None
+        self._on_socket_close = None
+        self._on_socket_register_write = None
+        self._on_socket_unregister_write = None
         self._websocket_path = "/mqtt"
         self._websocket_extra_headers = None
 
     def __del__(self):
         pass
 
-    def reinitialise(self, client_id="", clean_session=True, userdata=None):
-        if self._sock:
-            self._sock.close()
+    def _sock_recv(self, bufsize):
+        try:
+            return self._sock.recv(bufsize)
+        except socket.error as err:
+            if self._ssl and err.errno == ssl.SSL_ERROR_WANT_READ:
+                raise WouldBlockError()
+            if self._ssl and err.errno == ssl.SSL_ERROR_WANT_WRITE:
+                self._call_socket_register_write()
+                raise WouldBlockError()
+            if err.errno == EAGAIN:
+                raise WouldBlockError()
+            raise
+
+    def _sock_send(self, buf):
```

Comment:
I think that this addresses issue #188 as well.

Reply:
I don't really understand #188; I don't see the relation to my change. Maybe I looked at the wrong code lines?

Reply:
anonying.